### DISCUSSION TITLE
Tilføje data fra getUgeBreve request

### DISCUSSION THREAD

**Discussion by MartinLorenzen** [scaarup/aula#189] [Ideas]
Vil det være muligt at få tilføjet data fra denne request?
https://api.minuddannelse.net/json/metadata?op=GetUgebrevRequest

På vores børns skole får vi en ugentlig besked i weekenden med info om aktiviteter og ting der skal huskes til den kommende uge, og det ser ud til at være tilgængeligt via dette service kald.

Header eksempel:
```url
https://www.minuddannelse.net/api/stamdata/ugeplan/getUgeBreve?tidspunkt=2024-W43&elevId=3332957&_=1729877628839
```

Response eksempel:
```json
{
    "errorMessage": null,
    "ugebreve": [
        {
            "klasseNavn": "0D",
            "klasseId": 277724,
            "uge": 43,
            "indhold": "<p><strong>K&aelig;re alle!</strong></p><p><strong>Velkommen tilbage efter en forh&aring;bentlig god efter&aring;rsferie,</strong></p><p>Vi forts&aelig;tter i det nogenlunde s&aelig;dvanlige spor, men t&aelig;nker selvf&oslash;lgelig ogs&aring; p&aring;, at<strong> julen</strong> er noget ganske s&aelig;rligt i kommende periode - ikke mindst for b&oslash;rn.&nbsp;</p><p>Det vil vi selvf&oslash;lgelig g&oslash;re noget ud af. Herom senere!</p><p><strong>Idr&aelig;t</strong> vil forl&oslash;be som det<strong> plejer de n&aelig;ste 14 dage</strong> (uge 43 og 44).</p><p><strong>Fra uge 45</strong> kommer der en <strong>&aelig;ndring</strong> - idet b&oslash;rnene b&aring;de skal have <strong>skiftet&oslash;j og h&aring;ndkl&aelig;de med&nbsp;</strong>til idr&aelig;tstimerne. Jeg skal nok g&oslash;re opm&aelig;rksom p&aring; det igen i uge 44.</p><p><br></p><p><br></p>",
            "id": "85be9dce-92fa-4aac-b2ae-ad5e3c98112d"
        }
    ],
    "klasser": [
        {
            "skoleaarId": 2024,
            "id": 277724,
            "trin": 0,
            "harTrin": true,
            "navn": "0D"
        }
    ]
}
```

**Comment by scaarup** [comment]
Hej @MartinLorenzen 

Jo det kunne vi sikkert sagtens gøre. Men jeg tænker de her "UgeBreve" kommer med en bestemt widget. Lad mig lige se hvilke widgets du har til rådighed. Prøv et enable debug log og se efter linjen, som fx denne her:

`
2024-10-26 17:13:17.158 DEBUG (SyncWorker_0) [custom_components.aula.client] Widgets found: {'0069': 'KMD Educa Elev - Kontaktoplysninger', '0013': 'Nyttige links', '0029': 'MinUddannelse – Ugenoter', '0072': 'MU Elev - fravær'}
`

**Reply by MartinLorenzen** [reply]
Jeg har forsøgt at køre Chrome med debug logs slået til, som beskrevet i linket her, men det giver ikke noget output i stil med det du nævner.
https://support.google.com/chrome/a/answer/6271282?hl=en#zippy=%2Cmac

Jeg vil lige give det et skud mere en af aftenerne og så vender jeg tilbage.

**Reply by scaarup** [reply]
Nej det var bare på Home assistent integrationen jeg mente du skulle en have debug log og så kigge i ha loggen.

**Comment by MartinLorenzen** [comment]
Nu har jeg slået debug log til, men det eneste der kommer ud omkring widgets i aula integrationen er dette:

```log
2024-10-30 07:06:19.544 DEBUG (SyncWorker_63) [custom_components.aula.client] In the MU Opgaver flow
2024-10-30 07:06:19.544 DEBUG (SyncWorker_63) [custom_components.aula.client] Requesting new token for widget 0030
2024-10-30 07:06:19.694 DEBUG (SyncWorker_63) [custom_components.aula.client] MU Opgaver status_code 200
2024-10-30 07:06:19.694 DEBUG (SyncWorker_63) [custom_components.aula.client] MU Opgaver response {"opgaver":[{...}]}
```

**Reply by scaarup** [reply]
> Nu har jeg slået debug log til, men det eneste der kommer ud omkring widgets i aula integrationen er dette:
> 
> ```
> 2024-10-30 07:06:19.544 DEBUG (SyncWorker_63) [custom_components.aula.client] In the MU Opgaver flow
> 2024-10-30 07:06:19.544 DEBUG (SyncWorker_63) [custom_components.aula.client] Requesting new token for widget 0030
> 2024-10-30 07:06:19.694 DEBUG (SyncWorker_63) [custom_components.aula.client] MU Opgaver status_code 200
> 2024-10-30 07:06:19.694 DEBUG (SyncWorker_63) [custom_components.aula.client] MU Opgaver response {"opgaver":[{...}]}
> ```

ej det kan ikke passe :). Prøv lige at genstart HA og kig igen.

**Reply by MartinLorenzen** [reply]
Det virkede med en genstart. Så kommer følgende:

2024-10-30 12:08:58.884 DEBUG (SyncWorker_12) [custom_components.aula.client] Widgets found: {'0018': 'EasyIQ Links', '0072': 'MU Elev - fravær', '0023': 'MinUddannelse - SSO', '0030': 'MinUddannelse - Opgaver', '0029': 'MinUddannelse – Ugenoter', '0028': 'MinUddannelse – Forløb'}

**Comment by scaarup** [comment]
okay spændende. Jeg tænker det må være denne her:
'0028': 'MinUddannelse – Forløb'

Den har jeg nemlig ikke set før, eller implementeret. Jeg prøver at lave en test branch - vender retur.

**Reply by MichaelRyom** [reply]
Kom her for sammen :) Men det er vel 0029 (?) - Har ikke 0028 her og har adgang til "ugenoter/ugebrev"

2024-11-06 17:00:23.187 DEBUG (SyncWorker_8) [custom_components.aula.client] Widgets found: {'0030': 'MinUddannelse - Opgaver', '0029': 'MinUddannelse – Ugenoter', '0019': 'Biblioteket', '0057': 'Ugeplan', '0018': 'EasyIQ Links', '0047': 'Fravær - forældreindberetning', '0120': 'MinUddannelse - Meddelelsesbog'}
